apps/prairielearn/elements/pl-file-upload/pl-file-upload.js

```diff
@@ -15,6 +15,32 @@
       this.files = [];
       this.acceptedFiles = options.acceptedFiles || [];
       this.acceptedFilesLowerCase = this.acceptedFiles.map((f) => f.toLowerCase());
+      // The list of optional files contains tuples with patterns (if regex-based) and display names
+      this.optionalFiles = options.optionalFiles || [];
+      // Divide optional files into static names and regex patterns
+      this.optionalFilesStatic = this.optionalFiles.filter((f) => f[0] === '').map((f) => f[1]);
+      this.optionalFilesLowerCase = this.optionalFilesStatic.map((f) => f.toLowerCase());
+      this.optionalFilesWildcard = this.optionalFiles.filter((f) => f[0] !== '');
```

Comment:
Is there a reason why this division isn't done in Python and sent already divided here? You need to handle this distinction there anyway.

Also, not a huge fan of the use of tuples here, can you use dict/objects?

Reply:
That's basically an artifact of switching the format a couple of times while working out the best way to support both regex and static names. I agree this could be two parameters to make the code cleaner.

Reply:
Updated. I think changing this also helped clean up the Python code in the process.